src/measure.cpp

```diff
@@ -1047,6 +1059,11 @@ int Measure::AdjustXPos(FunctorParams *functorParams)
         StaffAlignment *staffAlignment = system->m_systemAligner.GetStaffAlignmentForStaffN(staffN);
         params->m_staffSize = (staffAlignment) ? staffAlignment->GetStaffSize() : 100;
 
+        // Prevent collisions of scoredef clefs with thick barlines
+        if (this->IsFirstInSystem() && system->GetDrawingScoreDef()->HasSystemStartLine()) {
+            params->m_upcomingMinPos = params->m_doc->GetDrawingBarLineWidth(params->m_staffSize);
+        }
```

Comment:
You do now want to check if the measure is the first of the system and if the scoreDef has a staff line for every staff in the measure. That will be quite inefficient for an orchestral score. Move this outside the loop and use a variable for resetting `m_upcomingMinPos`